### DISCUSSION TITLE
array/array

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#136] [Announcements]
# array/array

Java程序员进阶之路，小白的零基础Java教程，从入门到进阶，Java数组

https://tobebetterjavaer.com/array/array.html

**Comment by Cyril-1600** [comment]
哥, 您的 List<Integer> aList = Arrays.asList(anArray) 其實是無法的，因為 anArray 的型別是 int[]，但 List 的型別是 Integer，兩個不一樣

**Reply by itwanger** [reply]
收到，这部分已调整。

**Comment by 2258324319** [comment]
Arrays.asList的目标只能是对象，Arrays.asList(int[] arr)是不行的，必须是Arrays.asList(Integer[] arr)才能成功赋给List<Integer>
int[] 转 Integer[] 可以用 Arrays.stream(arr).boxed().toArray(Integer[]::new)
Java这个装箱机制用起来让我真的自闭，C#就不会这么折磨，可惜C#吕布骑狗。。。

**Reply by itwanger** [reply]
是的，想当然的忽略了。已修正。

**Reply by itwanger** [reply]
这种写法确实很恶心了，算是。对于初学者并不友好。

**Reply by itwanger** [reply]
这里贴一下，正确的应该是 `List<Integer> aList = Arrays.stream(anArray).boxed().collect(Collectors.toList());`

**Reply by CharlesShan-hub** [reply]
idea帮我简化了一下：

List<Integer> aList = Arrays.stream(new int[]{1,2,3,4,5}).boxed().toList();

**Comment by Cyril-1600** [comment]
感謝協助校正，謝謝你的幫助。

沉默王二 ***@***.***> 於 2023年2月22日 週三 上午9:31寫道：
